src/index.ts

```diff
@@ -36,8 +36,59 @@ export type NotarizeWaitOptions = NotarizeResult & NotarizeCredentials;
 export type NotarizeStapleOptions = Pick<NotarizeAppOptions, 'appPath'>;
 export type NotarizeOptions = NotarizeStartOptions;
 
-function authorizationArgs(opts: NotarizeCredentials): string[] {
-  if ('appleId' in opts) {
+function isPasswordCredentials(opts: NotarizeCredentials): opts is NotarizePasswordCredentials {
+  const creds = opts as NotarizePasswordCredentials;
+  return creds.appleId !== undefined || creds.appleIdPassword !== undefined;
+}
+
+function isApiKeyCredentials(opts: NotarizeCredentials): opts is NotarizeApiKeyCredentials {
+  const creds = opts as NotarizeApiKeyCredentials;
```

Comment:
```suggestion
function isApiKeyCredentials(creds: NotarizeApiKeyCredentials): boolean {;
```

Reply:
I can change this too. What are your thoughts on keeping the type guards? From what I read, the type predicate (`x is Type` syntax) in the handbook is recommended for this use case since `NotarizeApiKeyCredentials` is a union type:
```ts
export type NotarizeCredentials = NotarizePasswordCredentials | NotarizeApiKeyCredentials;
```

Reply:
OK, now that I've read the handbook reference, I agree that we should keep the type guards. I'd still rename the parameter.